app/src/app/[lng]/onboarding/setup/page.tsx

```diff
@@ -408,6 +434,16 @@ export default function OnboardingSetup({
 
   useEffect(() => {
     if (cityData) {
+      const population = cityData?.data.population.filter(
+        (item: any) => item.year === data.year,
+      );
+      const populationObject = {
+        year: population[0]?.year,
+        population: population[0]?.population,
+        datasourceId: population[0]?.datasource_id,
+      };
```

Comment:
Here you could use `.find` instead of `.filter`, so you don't have to index into the array afterwards and it stops after it has found the first matching entry.